### PR TITLE
Add inversion again (maybe more properly?)

### PR DESCRIPTION
**Pull request by ptwz**
Hi there,
this is another take on inversion.
Inversion can also be made optional, but that patch is not ready yet.

Please review, whether this patch is more appropriate.

### CODE REVIEW

File: zbar/processor.c [hunk 1]
```diff
@@ -219,7 +219,14 @@ int _zbar_process_image (zbar_processor_t *proc,
         }
         zbar_image_scanner_recycle_image(proc->scanner, img);
         int nsyms = zbar_scan_image(proc->scanner, tmp);
-        _zbar_image_swap_symbols(img, tmp);
+        if ( (nsyms == 0) && (1) ) //( proc->try_inversion ) )
```
[the remark this answers is not in the document]
Comment (ptwz) [reply zbar/processor.c]: Sorry, I was working on something like that. I did it a bit different. Having a flag in the processor.

File: zbar/processor.c [hunk 2]
```diff
@@ -219,7 +219,14 @@ int _zbar_process_image (zbar_processor_t *proc,
...
+        {
+            zbar_image_t *tmp_inverted = zbar_image_copy_invert(tmp);
+            nsyms = zbar_scan_image(proc->scanner, tmp_inverted);
+            _zbar_image_swap_symbols(img, tmp_inverted);
+            zbar_image_destroy(tmp_inverted);
+        } else
+            _zbar_image_swap_symbols(img, tmp);
```
[the remark this answers is not in the document]
Comment (ptwz) [reply zbar/processor.c]: Well, my first two attempts where in img_scanner. Those broke the cache which i could not wrap my head around.

File: include/zbar.h [hunk 3]
```diff
@@ -571,6 +571,11 @@ typedef void (zbar_image_data_handler_t)(zbar_image_t *image,
  */
 extern zbar_image_t *zbar_image_create(void);
 
+/** Create an inverted copy of an image.
+ * @returns a new image object with inverted content from source image
+ * @note This currently is only supported for 8bit greyscale images
+ */
+extern zbar_image_t *zbar_image_copy_invert(const zbar_image_t *);
```
[the remark this answers is not in the document]
Comment (ptwz) [reply include/zbar.h]: Ok, fair enough.

File: zbar/image.c [hunk 4]
```diff
@@ -212,6 +212,25 @@ zbar_image_t *zbar_image_copy (const zbar_image_t *src)
     return(dst);
 }
 
+zbar_image_t *zbar_image_copy_invert(const zbar_image_t *img)
+{
+    zbar_image_t *rw = zbar_image_copy(img);
```
[the remark this answers is not in the document]
Comment (ptwz) [reply zbar/image.c]: ok, maybe

**Comment by ptwz** [comment]
As i tried multiple iterations and seem not be able to fulfill the quality standards. I withdraw my request. Sorry to waste your time.

**Comment by mchehab** [comment]
> As i tried multiple iterations and seem not be able to fulfill the quality standards. I withdraw my request. Sorry to waste your time.

I would prefer if you keep it opened. If you don't have any time to address my comments, I may end importing your patches and do the needed changes on the top of it (when I have some time - should be busy those days handling some Kernel development).

**Pull request closed without merging**